front-end/src/main/kotlin/org/jetbrains/kotlin/spec/loader/GithubTestsLoader.kt

```diff
@@ -0,0 +1,139 @@
+package org.jetbrains.kotlin.spec.loader
+
+import js.externals.jquery.JQueryAjaxSettings
+import js.externals.jquery.JQueryPromise
+import js.externals.jquery.JQueryXHR
+import js.externals.jquery.`$`
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import org.jetbrains.kotlin.spec.entity.Section
+import org.jetbrains.kotlin.spec.entity.SectionTestMap
+import org.jetbrains.kotlin.spec.entity.test.Test
+import org.jetbrains.kotlin.spec.entity.test.TestPlace
+import org.jetbrains.kotlin.spec.entity.test.parameters.TestInfo
+import org.jetbrains.kotlin.spec.entity.test.parameters.testArea.TestArea
+import org.jetbrains.kotlin.spec.utils.format
+import kotlin.browser.window
+import kotlin.js.Promise
+
+interface GithubTestsLoader {
+    enum class TestFileType { SPEC_TEST, IMPLEMENTATION_TEST }
+
+    companion object {
+        private const val RAW_GITHUB_URL = "https://raw.githubusercontent.com/JetBrains/kotlin"
+
+        private const val SPEC_TEST_DATA_PATH = "compiler/tests-spec/testData"
+        private const val LINKED_SPEC_TESTS_FOLDER = "linked"
+
+        const val DEFAULT_BRANCH = "spec-tests"
+
+        fun getBranch() = window.localStorage.getItem("spec-tests-branch") ?: DEFAULT_BRANCH
+
+        fun loadTestFileFromRawGithub(
+                path: String,
+                testPathInSpec: String? = null,
+                customFolder: String? = null,
```

Comment:
What is the `customFolder` and what is the difference between `path` and `testPathInSpec`?

Reply:
customFolder and testPathInSpec are redundant. Removed.